tests/unit_tests/pyanaconda_tests/modules/subscription/test_subscription_tasks.py

```diff
@@ -936,33 +976,981 @@ def test_system_purpose_json_parsing(self):
+        )
+        conf_backup = task.run()
+        # check the RHSM config proxy was called correctly
+        config_proxy.GetAll.assert_called_once_with("")
+        # check the DBus struct is correctly converted to a Python dict
+        assert conf_backup == {"foo": "bar"}
+
+    def test_rhsm_roll_back(self):
+        """Test the RollBackSatelliteProvisioningTask."""
+        # create mock RHSM config proxy
+        config_proxy = Mock()
+        # and mock RHSM configuration
+        rhsm_config = {"foo": "bar"}
+        # create the task and run it
+        task = RollBackSatelliteProvisioningTask(
+            rhsm_config_proxy=config_proxy,
+            rhsm_configuration=rhsm_config
+        )
+        task.run()
+        # check the RHSM config proxy was called correctly
+        config_proxy.SetAll.assert_called_once_with({"foo": get_variant(Str, "bar")}, "")
+
+    @patch("pyanaconda.modules.subscription.satellite.run_satellite_provisioning_script")
+    def test_provision_target_no_op(self, run_script_function):
+        """Test the ProvisionTargetSystemForSatelliteTask - no op."""
+        # create the task and run it
+        task = ProvisionTargetSystemForSatelliteTask(provisioning_script=None)
+        task.run()
+        # make sure we did not try to provision the system with
+        # registered_to_satellite == False
+        run_script_function.assert_not_called()
+
+    @patch("pyanaconda.modules.subscription.satellite.run_satellite_provisioning_script")
+    def test_provision_target_success(self, run_script_function):
+        """Test the ProvisionTargetSystemForSatelliteTask - success."""
+        # make the run script function return True, indicating success
+        run_script_function.return_value = True
+        # create the task and run it
+        task = ProvisionTargetSystemForSatelliteTask(provisioning_script="foo")
+        task.run()
+        # make sure we did try to provision the system with
+        run_script_function.assert_called_once_with(
+            provisioning_script="foo",
+            run_on_target_system=True
+        )
+
+    @patch("pyanaconda.modules.subscription.satellite.run_satellite_provisioning_script")
+    def test_provision_target_failure(self, run_script_function):
+        """Test the ProvisionTargetSystemForSatelliteTask - failure."""
+        # make the run script function return False, indicating failure
+        run_script_function.return_value = False
+        # create the task and run it
+        task = ProvisionTargetSystemForSatelliteTask(provisioning_script="foo")
+        # check if the correct exception for a failure is raised
+        with pytest.raises(SatelliteProvisioningError):
+            task.run()
+        # make sure we did try to provision the system with
+        run_script_function.assert_called_once_with(
+            provisioning_script="foo",
+            run_on_target_system=True
+        )
+
+
+class RegisterandSubscribeTestCase(unittest.TestCase):
```

Comment:
a typo Registerand... -> RegisterAnd...
Feel free to ignore.

Reply:
Might as well fix it. :)